week_3/Binary_Tree/class.js

```diff
@@ -0,0 +1,117 @@
+class Node {
```

Comment:
you might want to consider renaming this file to "index.js"

That's a common way of expressing "this is the main file of this directory". Along side that directory, you could find other "util" files like the ones you have here (invert_binary_tree).

Also, it's good to be consistent with casing. Consider renaming `Binary_Tree` to `binary_tree` 😄 this makes it easier for other developers to import your code, and its often enforced via a linter in larger code bases.

Consider this paragraph, titled "Unsurprising", in Rust for Rustaceans. The chapter is describing creating code that's easy for others to use.

> The Principle of Least Surprise, otherwise known as the Law of Least Astonishment, comes up a lot in software engineering, and it holds true for Rust interfaces as well. Where possible, your interfaces should be intuitive enough that if the user has to guess, they usually guess correctly. Of course, not everything about your application is going to be immediately intuitive in this way, but anything that can be unsurprising should be. The core idea here is to stick close to things the user is likely to already know so that they don’t have to relearn concepts in a different way than they’re used to. That way you can save their brain power for figuring out the things that are actu- ally specific to your interface.
There are a variety of ways you can make your interfaces predictable. Here, we’ll look at how you can use naming, common traits, and ergonomic trait tricks to help the user out.